datafusion/core/src/physical_plan/explain.rs

```diff
@@ -121,8 +121,8 @@ impl ExecutionPlan for ExplainExec {
             )));
         }
 
-        let mut type_builder = StringBuilder::new(self.stringified_plans.len());
-        let mut plan_builder = StringBuilder::new(self.stringified_plans.len());
+        let mut type_builder = StringBuilder::new();
```

Comment:
👍 A lot of misuse of the previous version of the StringBuilder API: the capacity of StringBuilder is mistakenly taken as the number of items in the array.

How do you think if we use `StringBuilder::with_capacity(self.stringified_plans.len(), 1024)`, seems mirror the intentions clearer.

Reply:
So always `1024` when we don't know? Lack of knowledge of a good default was the only reason I didn't do that everywhere.

Reply:
FWIW I don't think the performance of explain plan is particularly important -- a few more allocations likely won't hurt either way

Reply:
I updated them all there it was sensible with a default of `1024`.